src/test.rs

```diff
@@ -0,0 +1,117 @@
+//! Test driver
+use crate::utils::ParseError;
+use std::str::FromStr;
+extern crate structopt;
+use structopt::StructOpt;
+mod basic;
+mod pfm;
+mod testutils;
+
+/// Test Struct
+pub struct Test {
+    pub name: String,
+    pub description: String,
+    pub call: fn() -> bool,
```

Comment:
I would piggyback on this comment and say that I think when a test fails it would ideally print to the CLI the specific error code it failed on, in addition to the FAILED (which does look very nice btw). Everything else I really like and would approve.

Reply:
Yeah, this is a way better idea. Thanks! I will go add this to the PR. I'll also have it print the exit code if it failed, and maybe more information too if a `-v, --verbose` flag is passed.